tools/bag_processing/scripts/rosbag_topic_filter.py

```diff
@@ -146,8 +154,21 @@ def topic_filter(verbose, inbag_path, accept_patterns, reject_patterns, outbag_p
         help="Print debug info",
         action="store_true",
     )
-    parser.add_argument("inbag", help="Input bag")
-    parser.add_argument("outbag", help="Filtered output bag")
 
     args = parser.parse_args()
-    topic_filter(args.verbose, args.inbag, args.accept, args.reject, args.outbag)
+
+    for inbag_path in args.inbag:
+        # Check if input bag exists
+        if not os.path.isfile(inbag_path):
+            print(("Bag file " + inbag_path + " does not exist."))
+            sys.exit()
+
+        output_bag_name = args.output.format(inbag=inbag_path)
```

Comment:
I like this way of forming the output bag name! However, if `inbag_path` is not just a bare filename but also specifies the directory, I think this won't do the desired thing. Ideally it could be split using `os.path.dirname()` /  `os.path.basename()` and just the basename part plugged into the output name template.